app.js

```diff
@@ -1,18 +1,65 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
 
 // require spotify-web-api-node package here:
+const SpotifyWebApi = require("spotify-web-api-node");
 
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
 
 // Our routes go here:
 
-app.listen(3000, () => console.log('My Spotify project running on port 3000 🎧 🥁 🎸 🔊'));
+app.get("/", (req, res) => {
+  res.render("index");
+});
+
+app.get("/artist-search", (req, res) => {
+  spotifyApi
+    .searchArtists(req.query.artist)
+    .then((data) => {
+      console.log(data.body.artists.items);
+      let allArtists = data.body.artists.items;
+      res.render("artist-search-results", { allArtists });
+    })
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+});
+
+app.get("/albums/:artistId", (req, res, next) => {
+  spotifyApi.getArtistAlbums().then(
```

Comment:
You need to do the `object destructuring`, in order to pass an argument to the `API method` you are using.
After that, after the promise is resolved, you will get the `data`, with the information you want. Then, you need to dig deeper into the `data` in order to filter the information that you want to send for the frontend. 
When you have the info that you want, you can store it inside a `variable`, and when rendering your `view`, pass the `variable` with the filtered data, in form of an `object`.
```
app.get('/albums/:artistId', (req, res, next) => {
  let { artistId } = req.params;
  spotifyApi.getArtistAlbums(artistId).then(
    (data) => {
      console.log('Artist albums', data.body.items);
      let allAlbums = data.body.items;
      res.render('albums', { allAlbums });
    },
    function (err) {
      console.error(err);
    }
  );
});
```